x-pack/plugins/cases/public/components/all_cases/solution_filter.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useState } from 'react';
+import {
+  EuiFilterButton,
+  EuiFilterSelectItem,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPanel,
+  EuiPopover,
+  EuiText,
+  EuiIcon,
+} from '@elastic/eui';
+import styled from 'styled-components';
+
+import * as i18n from './translations';
+import type { Solution } from './types';
+
+interface FilterPopoverProps {
+  onSelectedOptionsChanged: (value: string[]) => void;
+  options: Solution[];
+  optionsEmptyLabel?: string;
+  selectedOptions: string[];
+}
+
+const ScrollableDiv = styled.div`
+  max-height: 250px;
+  overflow: auto;
+`;
+
+const Icon = styled(EuiIcon)`
+  padding-right: 4px;
+`;
+
+const toggleSelectedGroup = (group: string, selectedGroups: string[]): string[] => {
+  const selectedGroupIndex = selectedGroups.indexOf(group);
+  if (selectedGroupIndex >= 0) {
+    return [
+      ...selectedGroups.slice(0, selectedGroupIndex),
+      ...selectedGroups.slice(selectedGroupIndex + 1),
+    ];
+  }
+  return [...selectedGroups, group];
+};
+
+/**
+ * Popover for selecting a field to filter on
+ *
+ * @param buttonLabel label on dropdwon button
+ * @param onSelectedOptionsChanged change listener to be notified when option selection changes
+ * @param options to display for filtering
+ * @param optionsEmptyLabel shows when options empty
+ * @param selectedOptions manage state of selectedOptions
+ */
+export const SolutionFilterComponent = ({
```

Comment:
What is the difference between this component and `FilterPopover`?

Reply:
Only `type` difference. SolutionFilter handles `Solution` type. 
So @jonathan-buttner suggested to move it into a separate component.
I also agreed that it makes sense to let `FilterPopover` handle only string options.

Reply:
Ok, that makes sense to me!